adm/src/main.rs

```diff
@@ -74,7 +74,13 @@ fn parse_args<'a>() -> ArgMatches<'a> {
                 (@arg start: --start +takes_value "the first block to list"))
             (@subcommand show =>
                 (about: "inspect a block in the blockstore")
-                (@arg block: +required "the block to show"))
+                (@arg identifier: +required "the identifier used to determine which bloc to show")
+                (@arg batch: --batch conflicts_with[transaction height]
```

Comment:
These arguments are a bit odd, in that you can't do "--batch=xxx" because these are flags.

Suggest -b,--block=xxx, -N,--block-num=xxx, -T,--transaction=xxx, -B,--batch=xxx, and identifier all be allowed with identifier being a search (you can differentate block-num from signatures based on length and digits) and you could search for block, batch, and then transaction id in that order.